client.go

```diff
@@ -133,8 +137,17 @@ func (c *client) Walk(ctx context.Context, fid Fid, newfid Fid, names ...string)
 
 	return rwalk.Qids, nil
 }
-
+func (c *client) MaxReadSize() int {
+	return c.transport.channel().MaxReadSize()
+}
+func (c *client) MaxWriteSize() int {
+	return c.transport.channel().MaxWriteSize()
+}
 func (c *client) Read(ctx context.Context, fid Fid, p []byte, offset int64) (n int, err error) {
+	if len(p) > c.transport.channel().MaxReadSize() {
```

Comment:
Problem is, the wording in the man pages seems to indicate that the client should not send message expecting that the response would be longer than msize. What we could do though, instead of failing, is to create a request respecting msize and return the actually read size